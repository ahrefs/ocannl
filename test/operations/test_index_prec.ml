open Base
module Ops = Ir.Ops
module Utils = Utils

let () =
  (* Test with default big_models=false, should use uint32 *)
  let idx_prec = Ops.index_prec () in
  Stdio.printf "With big_models=false, index precision: %s\n" 
    (Ops.prec_string idx_prec);
  assert (String.equal (Ops.prec_string idx_prec) "uint32");
  
  (* Test with big_models=true, should use uint64 *)
  let old_setting = Utils.settings.big_models in
  Utils.settings.big_models <- true;
  let idx_prec = Ops.index_prec () in
  Stdio.printf "With big_models=true, index precision: %s\n" 
    (Ops.prec_string idx_prec);
  assert (String.equal (Ops.prec_string idx_prec) "uint64");
  
  (* Restore the old setting *)
  Utils.settings.big_models <- old_setting;
  
  Stdio.printf "Index precision tests passed!\n"