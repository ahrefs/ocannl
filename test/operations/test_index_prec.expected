Retrieving commandline, environment, or config file variable ocannl_log_level
Found 0, in the config file
With big_models=false, index precision: uint32
With big_models=true, index precision: uint64
Index precision tests passed!